File: sdks/java/src/main/java/com/puxbay/resources/PurchaseOrdersResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.PurchaseOrder;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class PurchaseOrdersResource {
    private final Puxbay client;

    public PurchaseOrdersResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<PurchaseOrder> list(int page, String status) throws PuxbayException {
        String query = "purchase-orders/?page=" + page;
        if (status != null) query += "&status=" + status;
        return (PaginatedResponse<PurchaseOrder>) client.request("GET", query, null, PaginatedResponse.class);
    }

    public PurchaseOrder get(String poId) throws PuxbayException {
        return client.request("GET", "purchase-orders/" + poId + "/", null, PurchaseOrder.class);
    }

    public PurchaseOrder create(PurchaseOrder po) throws PuxbayException {
        return client.request("POST", "purchase-orders/", po, PurchaseOrder.class);
    }

    public PurchaseOrder update(String poId, PurchaseOrder po) throws PuxbayException {
        return client.request("PATCH", "purchase-orders/" + poId + "/", po, PurchaseOrder.class);
    }

    public PurchaseOrder receive(String poId, List<Map<String, Object>> items) throws PuxbayException {
        Map<String, Object> body = new HashMap<>();
        body.put("items", items);
        return client.request("POST", "purchase-orders/" + poId + "/receive/", body, PurchaseOrder.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/PurchaseOrdersResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.PurchaseOrder
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class PurchaseOrdersResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1, status: String? = null): PaginatedResponse<PurchaseOrder> {
        val statusParam = if (status != null) "&status=$status" else ""
        return client.request(HttpMethod.Get, "purchase-orders/?page=$page$statusParam")
    }
    
    suspend fun get(poId: String): PurchaseOrder =
        client.request(HttpMethod.Get, "purchase-orders/$poId/")
    
    suspend fun create(po: PurchaseOrder): PurchaseOrder =
        client.request(HttpMethod.Post, "purchase-orders/", po)
    
    suspend fun update(poId: String, po: PurchaseOrder): PurchaseOrder =
        client.request(HttpMethod.Patch, "purchase-orders/$poId/", po)
        
    suspend fun receive(poId: String, items: List<Map<String, Any>>): PurchaseOrder =
        client.request(HttpMethod.Post, "purchase-orders/$poId/receive/", mapOf("items" to items))
}
